src/main/java/step2/CalculatorString.java

```diff
@@ -0,0 +1,70 @@
+package step2;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CalculatorString {
+    private static final String DELIMITER_EMPTY = " ";
+    private static final String DELIMITER_COMMA = ",";
+    private static final String DELIMITER_COLON = ":";
+    private static final String DELIMITER_MIX = DELIMITER_COMMA + "|" + DELIMITER_COLON;
+
+    private final String text;
+    private final String delimiter;
+
+    private CalculatorString(String text, String delimiter) {
+        this.text = text;
+        this.delimiter = delimiter;
+    }
+
+    public static CalculatorString of(String text) {
+        if (text == null || text.trim().isEmpty()) {
+            return new CalculatorString(text, DELIMITER_EMPTY);
+        }
+
+        Matcher m = Pattern.compile("//(.)\n(.*)").matcher(text);
+
+        if (m.find()) {
+            return new CalculatorString(m.group(2), m.group(1));
+        }
+
+        if (text.indexOf(DELIMITER_COMMA) != -1 && text.indexOf(DELIMITER_COLON) != -1) {
+            return new CalculatorString(text, DELIMITER_MIX);
+        }
+
+        if (text.indexOf(DELIMITER_COMMA) != -1) {
```

Comment:
```suggestion
        if (text.contains(DELIMITER_COMMA)) {
```

좀 더 자연스러운 API를 사용할 수 있을 것 같아요 :)